httpcore5/src/main/java/org/apache/hc/core5/pool/LaxConnPool.java

```diff
@@ -481,12 +485,18 @@ public void release(final PoolEntry<T, C> releasedEntry, final boolean reusable)
                 }
             }
             else {
-                deallocatePoolEntry();
+                disposePoolEntry(releasedEntry);
             }
+            releaseSeqNum.incrementAndGet();
             servicePendingRequest();
         }
 
+
         private void servicePendingRequest() {
+            servicePendingRequests(false);
+        }
+
+        private void servicePendingRequests(final boolean wholeQueue) {
```

Comment:
@coxlinton Let's use an enum here instead of a boolean.

Reply:
- enum  QueueProcessingPolicy? QueueProcessingStrategy?
- PARTIAL vs. COMPLETE? 
- COMPLETE vs. INCOMPLETE?
- WHOLE vs. PARTIAL?